From: Thomas Klute <thomas2.klute@uni-dortmund.de>
Date: Tue, 28 Feb 2017 21:01:17 +0100
Subject: Wait for OCSP server to become available

---
 test/runtests | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/test/runtests b/test/runtests
index c1ff135..e7b10ed 100755
--- a/test/runtests
+++ b/test/runtests
@@ -198,7 +198,12 @@ if [ -n "${CHECK_OCSP_SERVER}" ]; then
 	store_ocsp="--outfile ${OCSP_RESPONSE_FILE}"
     fi
     echo "---- Testing OCSP server ----"
-    ocsptool --ask --nonce --load-issuer authority/x509.pem --load-cert server/x509.pem ${store_ocsp}
+    waited=0
+    until ocsptool --ask --nonce --load-issuer authority/x509.pem --load-cert server/x509.pem ${store_ocsp} \
+	    || [ ${waited} -ge 20 ]; do
+	sleep 1
+	waited=$((${waited} + 1))
+    done
     echo "---- OCSP test done ----"
 fi
 
